### PR TITLE
windows: Remove Unposix completely

### PR DESCRIPTION
**Pull request by JPTIZ**
Removes unposix completely, thus ensuring that no unistd.h or sys/time.h is being included in windows.

### CODE REVIEW

File: src/lib/eina/eina_bezier.c [hunk 1]
```diff
@@ -15,6 +15,8 @@
  * License along with this library;
  * if not, see <http://www.gnu.org/licenses/>.
  */
+#include <evil_private.h>
```
Comment (joaoantoniocardoso) [inline src/lib/eina/eina_bezier.c:18]:
```suggestion
#ifdef _WIN32
# include <evil_private.h>
#endif
```

File: src/lib/eina/eina_pipe.h [hunk 2]
```diff
@@ -1,17 +0,0 @@
-#ifndef EINA_PIPE_H
```
Comment (joaoantoniocardoso) [inline src/lib/eina/eina_pipe.h:1]: Why removing this file?

File: src/tests/eina/eina_test_value.c [hunk 3]
```diff
@@ -24,6 +24,14 @@
 #include <inttypes.h>
 #include <float.h>
 
+#ifdef _WIN32
+# ifndef WIN32_LEAN_AND_MEAN
+#  define WIN32_LEAN_AND_MEAN
+# endif
+# include <winsock2.h>
```
Comment (Coquinho) [inline src/tests/eina/eina_test_value.c:31]: Wasn't this already passing? Why now we need `winsock2.h` here?

File: src/lib/eina/eina_main.c [hunk 4]
```diff
@@ -27,7 +27,9 @@
 # ifdef HAVE_SYS_TYPES_H
 #  include <sys/types.h>
 # endif
-# include <unistd.h>
+# ifndef _WIN32
+#  include <unistd.h>
```
Comment (walac) [inline src/lib/eina/eina_main.c:31]: Whouldn't this change hit cygwin builds?

Comment (JPTIZ) [reply src/lib/eina/eina_main.c:31]: Indeed, didn't realize that. So I should change to `_MSC_VER`, right?

Comment (walac) [reply src/lib/eina/eina_main.c:31]: Ideally, we should use a combination of `_WIN32` and `__CYGWIN__` macros, but I think `_MSC_VER` does the job.

File: src/lib/eina/eina_value.c [hunk 5]
```diff
@@ -39,6 +48,8 @@
 #include "eina_safety_checks.h"
 #include "eina_value.h"
 
+#include <evil_private.h>
```
Comment (joaoantoniocardoso) [inline src/lib/eina/eina_value.c:51]: shouldn't it go inside _WIN32 guards?

**Pull request merged**